### DISCUSSION TITLE
Can someone explain the differences between export and build?

### DISCUSSION THREAD

**Discussion by subodh-malgonde** [vercel/next.js#12660] [Help] [answered]
I have developed a web app that uses Nextjs app for the frontend and a Python (Django) API server as the backend. Most of my front end pages have API calls to my backend server (in `ComponentDidMount` or in response to user actions like button clicks).

I want to deploy this app to my server. The backend deployment is sorted. I am using Nginx as the reverse proxy. 

I am confused about deploying the nextjs app. After reading the docs I figured there are 2 ways to do this:

1. Run `next build` and then `next start`. This will start a nodejs server on port 3000. I can channel traffic to this port using nginx.
2. Run `next export`. This will generate an `out` directory. I can channel incoming traffic to this directory using a reverse proxy like Nginx.

Which of the 2 options should I use? What are the differences?

**Comment by rafaelalmeidatk** [comment, marked as answer]
`next start` will start a Node server that is able to do all the data-fetching methods described in this page: https://nextjs.org/docs/basic-features/data-fetching

`next export` will export all your pages to static HTML files that you can serve with any host. This is similar to what create-react-app does, but you can still build dynamic pages at build-time with [`exportPathMap`](https://nextjs.org/docs/api-reference/next.config.js/exportPathMap).

**Comment by kevinolivar** [comment]
Hi there,

new to nextjs here. 

`next export` does the build first anyway. Therefore, why when deploying a nextjs app to the cloud we see examples using the following commands,

`next build & next export`

next export should suffice when deploying to a s3 bucket no? Seems redundant commands...

**Comment by russellpekala** [comment]
Wondering if there might be a way to use a combo next export / api routes for the following purpose: 

My company releases its app to both an electron app (which we use next export to create) and to the web.  Currently we have a backend API that both electron and the web-deployed app utilize.  

We would like to use next api routes to better organize our code and reduce load from our main API server.  Of course we understand that next export will not "build" the api routes into the exported html, but since we deploy to web as well, we are hoping we could somehow tell the static html that is exported to point to the web-deployed routes for API calls.  

Has anyone done this kind of pattern?

**Comment by PeterCastillo** [comment]
im trying to use next export in my next app, but i am having some issues while running the command
![image](https://github.com/vercel/next.js/assets/108549256/aa00273e-62e2-4f88-a491-8bf28395b365)
while looking at the cause of this, i found out that was for trying to access the session in server side, im using next-auth to manage authentication,  and i use getServerSession() to get the session in my page.tsx for fetching with the access token some data before showing the main components 
![image](https://github.com/vercel/next.js/assets/108549256/aaf9b559-1c3d-413a-a1c6-254bdaecc8a1)
